app/src/main/java/de/westnordost/streetcomplete/quests/smoking/AddSmoking.kt

```diff
@@ -0,0 +1,57 @@
+package de.westnordost.streetcomplete.quests.smoking
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.meta.updateWithCheckDate
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CITIZEN
+import de.westnordost.streetcomplete.ktx.arrayOfNotNull
+
+class AddSmoking : OsmFilterQuestType<SmokingAllowed>() {
+
+    private val elementFilterBasic = """
+        nodes, ways with
+        (
+          amenity ~ bar|cafe|pub|biergarten|restaurant|nightclub|stripclub
+          or leisure ~ outdoor_seating
+        )
+    """
+
+    override val elementFilter = elementFilterBasic + """
+	and (!smoking or smoking older today -8 years)
+    """
+
+    override val changesetComment = "Add smoking status"
+    override val wikiLink = "Key:smoking"
+    override val icon = R.drawable.ic_quest_smoke
+    override val isReplaceShopEnabled = true
+    override val defaultDisabledMessage = R.string.default_disabled_msg_go_inside_regional_warning
+
+    override val questTypeAchievements = listOf(CITIZEN)
+
+    override fun getTitle(tags: Map<String, String>) =
+        if (hasFeatureName(tags))
+            R.string.quest_smoking_name_type_title
+        else
+            R.string.quest_smoking_no_name_title
+
+    override fun getTitleArgs(tags: Map<String, String>, featureName: Lazy<String?>): Array<String> {
+        val name = tags["name"] ?: tags["brand"]
+        return arrayOfNotNull(name, featureName.value)
+    }
+
+    override fun getHighlightedElements(element: Element, getMapData: () -> MapDataWithGeometry) =
+        getMapData().filter(elementFilterBasic)
+
+    override fun createForm() = SmokingAllowedAnswerForm()
+
+    override fun applyAnswerTo(answer: SmokingAllowed, tags: Tags, timestampEdited: Long) {
+	tags.updateWithCheckDate("smoking", answer.osmValue)
+    }
+
+    private fun hasFeatureName(tags: Map<String, String>): Boolean =
+        tags.containsKey("name") || tags.containsKey("brand")
```

Comment:
I am not entirely sure is there a chance that correctly tagged `bar|cafe|pub|biergarten|restaurant|nightclub|stripclub` will have no `name` but will have `brand` tag.

Reply:
Neither did I, but apparently there [are a few](http://overpass-turbo.eu/?q=LyoKVGhpcyBoYcSGYmVlbiBnxI1lcmF0ZWQgYnkgdGhlIG92xJJwxIlzLXR1cmJvIHdpemFyZC7EgsSdxJ9yaWdpbmFsIHNlxLBjaMSsxIk6CsOiwoDCnGFtxI1pdMSafiAiYsSwfGNhZmV8cHVifGJpxJJnxLDElW58cmVzdGHEqGFudHxuxLhodGNsxZ18xarEt3DFtsWdIiDFrsSXxLvFjCDEhSBudWzEvcaBxJ9wxJLElG9yPSrFiMKdCiovCltvdXQ6anNvbl1bdGnFjMacxp4yNV07Ci8vxI_ElMSdciDFqHPGiXRzCigKICDGsCBxdcSSxJrEo3J0IGbGkTogxYjFisWMxbLFj34vxZRyxZbFmMWaxZzFnsWgxaLFpMSNxafFqcWrxa3Fr8WxxbPFtcW3xZ7FumnFvMepxrHGjMaDZSHGk8aAbsSXb8aOxJN0xpHGk8aVxr7Gh3dyWyLFi8WNxY8ifsWTxZXFl8WZxZvFuMedcsWjx4jHoMa3x6PEk8elxbJnxbTFvceqdMW7yJwixqQix7EiIciJLirIociDYsiXZMihKHt7YsSqeH19KcauCsi4xq_GsXDEt8WvxrbFqca5xrvGqcSYb2R5xq4-xq7JhXNrZcS9cXQ7&c=BO_juVVynJ). There is even more of those without `name` but with `operator`, so I've added that as fallback too: 1ae5f9d0c07ce230015d7fa3c5d9d7c334bb0f82